# Advanced Bioinformatics

## Table of Contents

* [Class 2: Introduction to the Linux command line, NGS data formats, read mapping and alignments](#class-2)
* [Hands On: Transcriptome Assembly: Case study of bacteria *Listeria monocytogenes*](#hands-on)
* [Docker Image](#docker-image)
* [Author](#author)
* [Contributing](#contributing)

## Docker Image

* [osvaldogc/ufv](https://hub.docker.com/r/osvaldogc/ufv/): Docker image with `workshop` installed.
Images can be pulled via `docker pull osvaldogc/ufv` 

## Running locally

In case you don't have Docker installed on your Linux machine. Try to install software required with 
next steps:

## Author

Fernando Pozo – [@fpozoca](https://twitter.com/fpozoca) – fpozoc@cnio.es

[Bioinformatics Unit](https://bioinformatics.cnio.es/) - [@BU_CNIO](https://twitter.com/BU_CNIO)
[Spanish National Cancer Research Centre (CNIO)](https://www.cnio.es/)

Distributed under the GNU General Public License. See ``LICENSE`` for more information.

[https://github.com/fpozoca](https://github.com/fpozoca)

## Contributing

1. Fork it (<https://github.com/fpozoc/advanced_bioinformatics>).
2. Create your feature branch (`git checkout -b feature/fooBar`).
3. Commit your changes (`git commit -am 'Add some fooBar'`).
4. Push to the branch (`git push origin feature/fooBar`).
5. Create a new Pull Request.
